### PR TITLE
Ts types

### PR DESCRIPTION
**Pull request by evenius**
Basically just restructuring erx to use Typescript, complete the typings and backport to flow!

### CODE REVIEW

File: src/promise.ts [hunk 1]
```diff
@@ -119,3 +130,22 @@ Promise.sequence = function sequence<A>(as: Array<Promise<A>>): Promise<Array<A>
     });
   });
 };
+
+Promise.all = Promise.sequence
```
Comment (HDykins) [inline src/promise.ts]: IS IT THOUGH?!

Comment (evenius) [reply src/promise.ts]: Let's see... It... Creates an unset array, walks over all the promises... Writes in place in the results-array as they're done... Returns when all are resolved... rejects immediately on Error... 

YES IT FUCKING WELL IS

**Comment by jimhbanks** [comment]
Hi

I contracted for you guys in 2016, and seem to be getting your github
messages (had not till last week)

You can remove me :-)

Jim Banks

**Comment by evenius** [comment]
Hey @jimhbanks!

Would you believe that this repository was last updated in 2016?

Anyway, the easiest way for you to no longer be notified with updated of this repo is to click the "👁 - Unwatch ▼" up there next to "Star" and "Fork".

Since this is a public repository I can't really seem to be able to moderate who's marked as a watcher, except for perhaps blocking you, so you could always try and write something _really insulting_... Although it might be for the best for all of us to just unwatch :)

Take care and have a lovely week!
// Jonathan Gill
Current contractor for adludio

**Comment by jimhbanks** [comment]
Ah I didnt see it was public :-) I'll unwatch and leave you to it!

All the best

Jim

**Pull request merged**